source/fundamentals/data-formats/documents.txt

```diff
@@ -14,34 +14,351 @@ Documents
 Overview
 --------
 
-- Definition - what is a document, what format is it, where to find more info 
-- Table comparison of classes (columns for: implements Map, provides type safety, is an interface or class, compatibility with server versions
-- Recommend Document for general use and why.
-- Duplicate field name admonition (I forgot what this is)
+A MongoDB **document** is a data structure that contains key/value pairs in
+binary JSON (BSON) format and is the base unit in the MongoDB data hierarchy.
+
+Each **document** belongs to a single **collection** and each **collection**
+belongs to a single **database**. For more information, see our MongoDB manual
+entry on :manual:`Documents </core/document>`.
+
+The Java driver and BSON library include the following document classes that
+help you access and manipulate the BSON data in documents:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 10 10 20 20
+
+   * - Name
+     - Package
+     - Implements Map
+     - Recommended Usage
+
+   * - ``Document``
+     - ``org.bson``
+     - Yes, implements ``Map<String, Object>``
+     - When you want a flexible and concise data representation.
+
+   * - ``BsonDocument``
+     - ``org.bson``
+     - Yes, implements ``Map<String, BsonValue>``
+     - When you need a type-safe API.
+
+   * - ``BasicDBObject``
+     - ``com.mongodb``
+     - No
+     - When you are migrating your application from a legacy driver version.
+
+While you can use any of the document classes in your application, we
+recommend that you use the ``Document`` class since it can concisely
+represent dynamically structured documents of any complexity. It
+implements the ``Map<String, Object>`` interface which enables it to use
+loosely-typed values.
+
+.. warning:: Duplicate Field Names
+
+   Although a document is valid if it contains two duplicate keys, the driver
+   only sets the value of the key to the value in the last instance.
```

Comment:
I don't have any qualms with adding this info, but "field" is included in the linked doc: https://docs.mongodb.com/manual/core/document/#field-names

I'll make sure I explain that the link has important info about field-related terminology and limitations, and remove this warning. I think the reason it is called out in the current version of the driver docs is because they don't link to the manual entry: https://mongodb.github.io/mongo-java-driver/4.1/bson/documents/